jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -82,6 +87,17 @@ class JobRunExecutorActor(
     context.become(starting)
   }
 
+  def scheduleStartingDeadlineTimeout(): Unit = {
+    import scala.concurrent.duration._
+
+    startingDeadline.foreach { deadline =>
+      val now = clock.now()
+      val from = run.createdAt.plus(deadline.toMillis)
+      val timeout = Seconds.secondsBetween(now, from).getSeconds.seconds
+      context.system.scheduler.scheduleOnce(timeout, self, StartTimeout)
```

Comment:
This has to be cancellable and needs to be cancelled on `start` :)
example:  https://github.com/dcos/metronome/blob/master/jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecSchedulerActor.scala#L27
and
https://github.com/dcos/metronome/blob/master/jobs/src/main/scala/dcos/metronome/jobspec/impl/JobSpecSchedulerActor.scala#L73

Reply:
1. I'm not sure `createdAt` is Some at this point...   is this not None for the first launch at preStart?
2. if createdAt is None, then use now for calculation
3. Love the use of createdAt... however the timeout could be negative if the time has past.   We should check for that and abort if true